nima/tests/integration/webclient/webclient/src/test/java/io/helidon/nima/webclient/http1/ClientRequestImplTest.java

```diff
@@ -0,0 +1,410 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.nima.webclient.http1;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.UncheckedIOException;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.ArrayList;
+
+import io.helidon.common.GenericType;
+import io.helidon.common.http.Headers;
+import io.helidon.common.http.Http;
+import io.helidon.common.http.WritableHeaders;
+import io.helidon.nima.http.media.EntityReader;
+import io.helidon.nima.http.media.EntityWriter;
+import io.helidon.nima.http.media.MediaContext;
+import io.helidon.nima.webclient.ClientConnection;
+import io.helidon.nima.webclient.WebClient;
+import io.helidon.nima.webclient.http1.ClientRequestImpl;
+import io.helidon.nima.webserver.WebServer;
+import io.helidon.nima.webserver.http.ServerRequest;
+import io.helidon.nima.webserver.http.ServerResponse;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import static io.helidon.common.testing.http.junit5.HttpHeaderMatcher.hasHeader;
+import static io.helidon.common.testing.http.junit5.HttpHeaderMatcher.noHeader;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.IsNot.not;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class ClientRequestImplTest {
+    private static final Http.HeaderValue REQ_CHUNKED_HEADER = Http.Header.createCached(
+            Http.Header.create("X-Req-Chunked"), "true");
+    private static final Http.HeaderValue REQ_EXPECT_100_HEADER_NAME = Http.Header.createCached(
+            Http.Header.create("X-Req-Expect100"), "true");
+    private static final Http.HeaderName REQ_CONTENT_LENGTH_HEADER_NAME = Http.Header.create("X-Req-ContentLength");
+    private static final long NO_CONTENT_LENGTH = -1L;
+    private static final Http1Client client = WebClient.builder().build();
+    private static WebServer webServer;
+    private static int port;
+
+    @BeforeAll
+    static void beforeAll() {
+        webServer = WebServer.builder()
+                .routing(router -> router
+                        .put("/test", ClientRequestImplTest::responseHandler)
+                        .put("/chunkresponse", ClientRequestImplTest::chunkResponseHandler)
+                )
+                .build()
+                .start();
+        port = webServer.port();
+    }
+
+    @AfterAll
+    static void afterAll() {
+        if (webServer != null) {
+            webServer.stop();
+        }
+    }
+
+    // @Test
```

Comment:
There are lots of tests commented out here

Reply:
Currently testing a failure that only happens in the pipeline but unable to reproduce locally. The code will still change in the coming days as I conduct the changes.